src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -297,6 +298,74 @@ public AgentSettings LoadSettings()
             }
         }
 
+        public async Task UnconfigureAsync(CommandSettings command)
+        {
+            if (!_store.IsConfigured())
+            {
+                _term.WriteError(StringUtil.Loc("MissingSettings"));
+            }
+
+            if (!_store.HasCredentials())
+            {
+                _term.WriteError(StringUtil.Loc("MissingCredentials"));
+            }
+
+            //stop, uninstall service and remove service config file
+            if (_store.IsServiceConfigured())
+            {
+                var serviceControlManager = HostContext.GetService<IServiceControlManager>();
+                try
+                {
+                    serviceControlManager.UnconfigureService();
+                    _term.WriteLine(StringUtil.Loc("ServiceUninstalled"));
+                }
+                catch (Exception ex)
+                {
+                    Trace.Error(ex);
+                    _term.WriteError(StringUtil.Loc("ServiceUninstallFailed"));
+                }
+            }
+
+            //delete agent from the server
+            try
+            {
+                AgentSettings settings = _store.GetSettings();
+                var credentialManager = HostContext.GetService<ICredentialManager>();
+                VssCredentials creds = credentialManager.LoadCredentials();
+                Uri uri = new Uri(settings.ServerUrl);
+                VssConnection conn = ApiUtil.CreateConnection(uri, creds);
+                var agentSvr = HostContext.GetService<IAgentServer>();
+                await agentSvr.ConnectAsync(conn);
+                await agentSvr.DeleteAgentAsync(settings.PoolId, settings.AgentId);
+                _term.WriteLine(StringUtil.Loc("AgentDeleted"));
+            }
+            catch (Exception ex)
+            {
+                Trace.Error(ex);
+                _term.WriteError(StringUtil.Loc("DeleteAgentFailed"));
+            }
+
+            //delete credential config file
+            try
+            {
+                _store.DeleteCredential();
+            }
+            catch (Exception ex)
+            {
+                Trace.Error(ex);
+            }
+
+            //delete settings config file
+            try
+            {
+                _store.DeleteSettings();
+            }
+            catch (Exception ex)
+            {
+                Trace.Error(ex);
+            }
+        }
```

Comment:
This is the latest after offline talk: I'll change the error handling to hard stop after an exception (instead of running all steps) and re-throw to the parent. A localized message will be printed, while the parent function will print the exception message.